projects/netgrif-components-core/src/lib/navigation/utility/filter-extraction.service.ts

```diff
@@ -49,16 +51,23 @@ export class FilterExtractionService {
         return navigationItemTaskAllowedNetsServiceFactory(this._factory, this.baseAllowedNets, sliced)
     }
 
-    public extractCompleteAdditionalFilterFromData(dataSection: Array<DataGroup>): Filter | undefined {
+    public extractCompleteAdditionalFilterFromData(activatedRoute: ActivatedRoute, dataSection: Array<DataGroup>): Filter | undefined {
         const taskRefIndex = getFieldIndexFromDataGroups(dataSection, GroupNavigationConstants.ITEM_FIELD_ID_ADDITIONAL_FILTER_TASKREF);
         if (taskRefIndex === undefined) {
             return undefined;
         }
 
-        return this.extractCompleteFilterFromData(dataSection.slice(taskRefIndex.dataGroupIndex + 1));
+        return this.extractCompleteFilterFromData(activatedRoute, dataSection.slice(taskRefIndex.dataGroupIndex + 1));
     }
 
-    public extractCompleteFilterFromData(dataSection: Array<DataGroup>, fieldId: string = UserFilterConstants.FILTER_FIELD_ID): Filter | undefined {
+    public extractCompleteFilterFromData(activatedRoute: ActivatedRoute, dataSection?: Array<DataGroup>, fieldId: string = UserFilterConstants.FILTER_FIELD_ID): Filter | undefined {
```

Comment:
can be activatedRoute also the optional ? i dont think we want to put activated route in this function on views that arent supposed to take route parameter